crates/services/sync/src/import.rs

```diff
@@ -423,34 +383,27 @@ fn check_sealed_header<
 >(
     header: &SealedBlockHeader,
     peer_id: PeerId,
-    p2p: Arc<P>,
-    consensus: Arc<C>,
-) -> Result<(), ImportError> {
+    p2p: &Arc<P>,
+    consensus: &Arc<C>,
+) -> bool {
     let validity = consensus
         .check_sealed_header(header)
-        .map_err(ImportError::ConsensusError)
-        .trace_err("Failed to check consensus on header")?;
-    if validity {
-        Ok(())
-    } else {
-        report_peer(
-            p2p.as_ref(),
-            peer_id.clone(),
-            PeerReportReason::BadBlockHeader,
-        );
-        Err(ImportError::BadBlockHeader)
+        .trace_err("Failed to check consensus on header")
+        .unwrap_or(false);
+    if !validity {
+        report_peer(p2p, peer_id.clone(), PeerReportReason::BadBlockHeader);
     }
+    validity
 }
 
 async fn await_da_height<C: ConsensusPort + Send + Sync + 'static>(
     header: &SealedBlockHeader,
-    consensus: &C,
-) -> Result<(), ImportError> {
-    consensus
+    consensus: &Arc<C>,
+) {
+    let _ = consensus
         .await_da_height(&header.entity.da_height)
         .await
-        .map_err(ImportError::ConsensusError)?;
-    Ok(())
+        .trace_err("Failed to wait for DA layer to sync");
```

Comment:
It looks like you're changing the function signature to be infallible by ignoring any internal error. Is this safe? What does it mean if `consensus.await_da_height(..)` fails for some reason, but the import continues?

Reply:
Yes, we continue to import blocks. If `consensus.await_da_height` fails, it may fail the execution of some blocks with new messages.

But it is not a huge problem. The block producer's public key is defined in the Genesis block, so awaiting the right DA block height is not critical for us if the block producer signed this block.